kwok/tests/usersignup_test.go

```diff
@@ -0,0 +1,95 @@
+package kwok
+
+import (
+	"encoding/json"
+	"os/exec"
+	"testing"
+
+	"github.com/redhat-appstudio/segment-bridge.git/containerfixture"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	contextName      = "kwok"
+	clusterServerURL = "http://127.0.0.1:8080"
+	scriptPath       = "../../scripts/get-uid-map.sh"
+)
+
+func TestGetUIDMap(t *testing.T) {
+	assert := assert.New(t)
+
+	deployment, err := setUpClusterConfiguration()
+	assert.NoError(err, "Failed to set up cluster configuration")
+	t.Cleanup(func() {
+		containerfixture.Cleanup(deployment["containerID"],
+			deployment["podName"])
+	})
+
+	cmd := exec.Command("/bin/sh", scriptPath)
+	output, err := cmd.CombinedOutput()
+	assert.NoError(err, "Failed to execute command\nOutput:", string(output))
+
+	m := make(map[string]int64)
+	err = json.Unmarshal(output, &m)
+	assert.NoError(err, "Failed to parse result")
+
+	t.Run("Usersignup validation", func(t *testing.T) {
+		assert.True(validateMap(m),
+			"Returned Usersignup map was incorrect, got:", m)
+	})
+
+	t.Run("Correct validation", func(t *testing.T) {
+		assert.True(validateMap(map[string]int64{
+			"petta": 12345678,
+			"blee1": 87654321,
+		}), "Correct map should be accepted")
+	})
+
+	t.Run("empty key", func(t *testing.T) {
+		assert.False(validateMap(map[string]int64{
+			"": 12345678,
+		}), "Map with an empty key should be rejected")
+	})
+
+	t.Run("no value key", func(t *testing.T) {
+		assert.False(validateMap(map[string]int64{
+			"<no value>": 87654321,
+		}), "Map with '<no value>' key should be rejected")
+	})
+
+	t.Run("empty map", func(t *testing.T) {
```

Comment:
I see multiple test-cases for validateMap, why not use the parametrize tests (want/got with a struct of test cases)?